src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -283,6 +288,24 @@ We support
                         variants.Add(variant);
                     }
                 }
+
+                telemetryEnabled = configurationSection.GetValue<bool>("TelemetryEnabled");
+
+                IConfigurationSection telemetryMetadataSection = configurationSection.GetSection("TelemetryMetadata");
```

Comment:
TelemetryMetadata is meant to be open ended. Customer should be able to put whatever <string, string> pairs they want to flow to telemetry destination.

Reply:
Where would they define items in it? Currently the provider is grabbing specific fields and putting them into the TelemetryMetadata object.

Reply:
should grab everything inside of TelemetryMetadata.

Reply:
```
foreach (section in telemetryMetadataSection.GetChildren())
{
    telemetryMetadata.Add(section.Key, section.Value);
}
```

Reply:
Is TelemetryMetadata something that will show up in the portal?

I get what you're saying, and we can do that within FeatureManagement, but we should separate "well defined" fields that the provider is persisting (like ETag and Label) from the every other field a user could add. Otherwise there could be collisions

Reply:
> Is TelemetryMetadata something that will show up in the portal?

I don't expect so.

Customers already can set tags on the feature flag.

Reply:
For customers using the portal, they would have no place to put things to be injected into TelemetryMetadata (besides within tags). Changing it to a dictionary would only enable custom telemetry fields in the local configuration case.

It could be relevant for the local json case, but I don't think it justifies removing the types for the other fields.

Reply:
This library knows nothing about etag, label, tags etc. All it knows is it can send telemetry.

Reply:
Synced with Jimmy. I was converted. Updated to expect a flattened "TelemetryMetadata" field on the FeatureFlag schema

Reply:
The TelemetryMetadata is an extensibility for feature flag providers (sources). It is NOT for end users. It's up to each feature flag provider to decide how to expose this capability to their end users or not expose it at all.

App Config, as one of the feature flag providers, will expose this capability to its end users via Label, ETag, and Tags. Configuration file is another type of feature flag provider and it may decide not to expose this capability to its end users at all.